protoc-gen-grpc-gateway/main.go

```diff
@@ -22,12 +22,14 @@ import (
 )
 
 var (
-	importPrefix         = flag.String("import_prefix", "", "prefix to be added to go package paths for imported proto files")
-	importPath           = flag.String("import_path", "", "used as the package if no input files declare go_package. If it contains slashes, everything up to the rightmost slash is ignored.")
-	registerFuncSuffix   = flag.String("register_func_suffix", "Handler", "used to construct names of generated Register*<Suffix> methods.")
-	useRequestContext    = flag.Bool("request_context", true, "determine whether to use http.Request's context or not")
-	allowDeleteBody      = flag.Bool("allow_delete_body", false, "unless set, HTTP DELETE methods may not have a body")
-	grpcAPIConfiguration = flag.String("grpc_api_configuration", "", "path to gRPC API Configuration in YAML format")
+	importPrefix              = flag.String("import_prefix", "", "prefix to be added to go package paths for imported proto files")
+	importPath                = flag.String("import_path", "", "used as the package if no input files declare go_package. If it contains slashes, everything up to the rightmost slash is ignored.")
+	registerFuncSuffix        = flag.String("register_func_suffix", "Handler", "used to construct names of generated Register*<Suffix> methods.")
+	useRequestContext         = flag.Bool("request_context", true, "determine whether to use http.Request's context or not")
+	allowDeleteBody           = flag.Bool("allow_delete_body", false, "unless set, HTTP DELETE methods may not have a body")
+	grpcAPIConfiguration      = flag.String("grpc_api_configuration", "", "path to gRPC API Configuration in YAML format")
+	allowRepeatedFieldsInBody = flag.Bool("allow_repeated_fields_in_body", false, "allows to use repeated field in `body` field of `google.api.http` annotation")
+	useJSONNameInSwaggerDef   = flag.Bool("json_name_in_swgdef", false, "if it sets Field.GetJsonName() will be used for generating swagger definitions, otherwise Field.GetName() will be used")
```

Comment:
Why is this flag only affecting the swagger definition? Won't this break the api generated with `protoc-gen-grpc-gateway`?

If this affects only `protoc-gen-swagger`, why is this a flag in `protoc-gen-grpc-gateway`?

Reply:
You're right, this flag should be in protoc-gen-swagger, i'll move it.

Reply:
>2. Please consider the very, very probable outcome where you won't be able to merge the upstream google.api.HttpRule change -- can you reserve a new set of proto options in the option registry for grpc-gateway (in addition to the existing swagger-specific ones) and implement this using that?

@ivucica do you mean implement option like this:
```
message AccountArray {
    repeated Account account = 1 [(grpc.gateway.options).unwrap_field = true]; // or promote_field?
}
```

I think it's less obvious solution than `response_body` field, and in this case you can't add additional bindings with different requests/responses mapping for example, how do you implement this with you solution?

```
service Strings {
    rpc ToUpper (String) returns (String) {
        option (google.api.http) = {
            post: "/strings/to_upper"
            body: "str"
            response_body: "str"
            additional_bindings: {
                post: "/strings/to_upper/v2",
                body: "*",
                response_body: "*",
            }
        };
    }
}

message String {
    repeated string str = 1;
}
```

Reply:
That was my original proposal, but I am fine with attaching it to `rpc` instead of `message`. We only need to move it to a proto field we actually control.

It is correct that you cannot add a different `response_body` with what I am proposing:

```
service Strings {
    rpc ToUpper (String) returns (String) {
        option (google.api.http) = {
            post: "/strings/to_upper"
            body: "str"
            additional_bindings: {
                post: "/strings/to_upper/v2",
                body: "*",
            }
        };
        option (grpc.gateway.rpc_options) = {
            response_body: "str"
        };
    }
}
```

That is indeed unfortunate, but I see `additional_bindings` as a bonus, and for the most part, I just use a 1:1 mapping.

Anything I can think of is ugly, including this possibly tolerable addition (which is not incompatible with what's above):

    rpc ToUpper (String) returns (String) {
        option (google.api.http) = {
            post: "/strings/to_upper"
            body: "str"
            additional_bindings: {
                post: "/strings/to_upper/v2",
                body: "*",
            }
        };
        option (grpc.gateway.rpc_options) = {
            configure_rule: {
                rule: { post: "/strings/to_upper" }
                config: { response_body: "str" }
            }
            configure_rule: {
                rule: { post: "/strings/to_upper/v2" }
                config: { response_body: "*" }
            }
        };
    }


To get what you want to happen, we _do_ need to get upstream to release the `response_body`, which may be difficult.

Once again: I am nearly certain your `.HttpRule` PR will not be accepted, due to insights that I cannot detail here without approval. So, once again: you should explore alternative designs unless upstream decides to provide the field.

I strongly suspect `grpc-gateway` will not move to a forked `google.api.HttpRule` either (especially one that may long-term conflict with upstream developments).

Reply:
I propose we take design discussions into #707 instead of this particular PR?